src/ucai/core/databricks.py

```diff
@@ -89,22 +92,27 @@ def extract_function_name(sql_body: str) -> str:
     Extract function name from the sql body.
     CREATE FUNCTION syntax reference: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-sql-function.html#syntax
     """
+    # NOTE: catalog/schema/function names follow guidance here:
+    # https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html#catalog-name
     pattern = re.compile(
         r"""
         CREATE\s+(?:OR\s+REPLACE\s+)?      # Match 'CREATE OR REPLACE' or just 'CREATE'
         (?:TEMPORARY\s+)?                  # Match optional 'TEMPORARY'
         FUNCTION\s+(?:IF\s+NOT\s+EXISTS\s+)?  # Match 'FUNCTION' and optional 'IF NOT EXISTS'
-        ([\w.]+)                           # Capture the function name (including schema if present)
+        ([^ /.]+\.[^ /.]+\.[^ /.]+)          # Capture the function name (including schema if present)
```

Comment:
In a follow-up, can we use named groups: https://docs.python.org/3.12/library/re.html?